### ISSUE TITLE
Prohibit empty parentheses in new-parens rule

### ISSUE THREAD

**Issue by powelski** [eslint/eslint#8421]
Currently, `new-parens` rule allows to enforce parentheses usage.

Some developers, including myself, prefer to enforce *skipping* of parentheses for `new Something`, whenever there are no constructor arguments. It would be nice to have an option to prevent `new Something()` in favor of `new Something`.

Current behavior:
```json
{
    "rules": {
        "new-parens": "error"
    }
}
```

```javascript
var cat1 = new Cat; // Missing '()' invoking a constructor. (new-parens)
var cat2 = new Cat(); // No error
```

Expected behavior example:
```json
{
    "rules": {
        "new-parens": ["error", "never"]
    }
}
```

```javascript
var cat1 = new Cat; // No error
var cat2 = new Cat(); // Unexpected '()' invoking a constructor. (new-parens)
```

**Comment by eslintbot** [comment]
Hi @powelski, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Hi @powelski, thanks for the issue. new-parens is one of our older rules that was lifted (more or less) straight from JSHint, so it could probably use an enhancement.

Would you mind adjusting your post to follow our [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md) please? It will help us evaluate your proposal and I greatly appreciate you taking the time. Thanks!

**Comment by powelski** [comment]
I provided the examples. Please let me know if it's enough now :)

**Comment by Aladdin-ADD** [comment]
please follow our [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md). thanks~ 😃

**Comment by vitorbal** [comment]
Thanks, @powelski! Would you be willing to submit a PR for this change if it gets accepted?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
